src/streamlink/plugins/wwenetwork.py

```diff
@@ -21,7 +20,10 @@
 
 
 @pluginmatcher(re.compile(
-    r"https?://watch\.wwe\.com/(channel)?",
+    r"https?://network\.wwe\.com/video/(?P<stream_id>\d+)?",
+))
+@pluginmatcher(re.compile(
+    r"https?://network\.wwe\.com/live/(?P<stream_id>\d+)?",
```

Comment:
Why are both capture groups optional? This results in `stream_id` being `None` if unset. This shouldn't match.

Btw, if you set a name for each matcher, then you can easily check which one matched and check for either live streams or VODs respectively. Using different capture group names would achieve the same, but using matcher names is much cleaner. But as said, the plugin requires a proper rewrite anyway (with validation schemas, etc), so let's not over-complicate things here for this simple fix.